openghg/plotting/_timeseries.py

```diff
@@ -95,28 +227,64 @@ def plot_timeseries(
             unit_conversion = float(data_units) / float(unit_value)
         else:
             unit_conversion = 1
+            # TODO: Not sure what is expected for unit_value here
+            unit_value = "1"
 
         y_data *= unit_conversion
 
-        fig.add_trace(go.Scatter(name=legend, x=x_data, y=y_data, mode="lines"))
+        unit_string = attributes_data["unit_print"][unit_value]
 
-    return fig
+        # Determine whether data is ascending or descending (positioning of legend)
+        ny = len(y_data)
+        data_left = y_data.isel(time=slice(int(ny * 0.9), -1)).mean()
+        data_right = y_data.isel(time=slice(0, int(ny * 0.1))).mean()
+        if data_left > data_right:
+            ascending.append(True)
+        else:
+            ascending.append(False)
 
+        # Add NaNs where there are large data gaps
+        x_data_plot, y_data_plot = _plot_remove_gaps(x_data.values, y_data.values)
 
-# def plot_timeseries(
-#     data: Union[ObsData, List[ObsData]], xvar: str, yvar: str, xlabel: str, ylabel: str
-# ) -> None:
-#     """Plot timeseries data using Plotly
+        # Convert unit string to html
+        unit_string_html = _latex2html(unit_string)
 
-#     Args:
-#         data: One or more ObsData objects
-#         xvar: Variable to plot on x axis
-#         yvar: Variable to plot on y axis
-#         xlabel: Label for x axis
-#         ylabel: Label for y axis
-#     Returns:
-#         plot type
-#     """
+        # Create plot
+        fig.add_trace(
+            go.Scatter(
+                name=legend_text,
+                x=x_data_plot,
+                y=y_data_plot,
+                mode="lines",
+                hovertemplate="%{x|%Y-%m-%d %H:%M}<br> %{y:.1f} " + unit_string_html
+            )
+        )
 
+        # Save units and species names for axis labels
+        unit_strings.append(unit_string_html)
+        species_strings.append(species_string)
 
-#     for dat in data:
+    if len(set(unit_strings)) > 1:
+        raise NotImplementedError("Can't plot two different units yet")
+
+    # Write species and units on y-axis
+    if ylabel:
```

Comment:
Here we should check `if ylabel is not None`

Reply:
Updated this